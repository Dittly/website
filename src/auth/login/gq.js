import {gql} from 'react-apollo'

export const signIn = gql`
mutation Signin($email: String!, $password: String!) {
  signinUser(email: { email: $email, password: $password }) {
    token
  }
}
`
