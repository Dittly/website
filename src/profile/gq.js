import {gql} from 'react-apollo'

export const getAllProjects = gql`
  query getAllProjects($userId: ID!) {
    allProjects(filter: {user: {id: $userId}}, orderBy: createdAt_DESC) {
      id
      title
      location
      type
      user {
        id
        name
      }
    }
  }
`
