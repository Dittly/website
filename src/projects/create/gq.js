import gql from 'graphql-tag'

export const getProjectTypes = gql`
  query enumValuesOfMetaInformationTags {
    __type(name: "ProjectTypes") {
      name
      enumValues {
        name
      }
    }
  }
`

export const createProject = gql`
  mutation createProjectMutation(
    $title: String!
    $location: String!
    $type: ProjectTypes!
    $userId: ID!
  ) {
    createProject(
      title: $title
      location: $location
      type: $type
      userId: $userId
    ) {
      id
      title
      location
      type
      user {
        id
        name
        email
      }
    }
  }
`
