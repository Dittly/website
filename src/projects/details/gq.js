import {gql} from 'react-apollo'

export const getProject = gql`
  query getProject($projectId: ID!) {
    Project(id: $projectId) {
      id
      title
      location
      type
      user {
        id
        name
      }
    }
  }
`
