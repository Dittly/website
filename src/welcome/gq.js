import gql from 'graphql-tag'

export const getAllProjects = gql`
  query getAllProjects {
    allProjects(orderBy: createdAt_DESC) {
      id
      title
      location
      type
      user {
        id
        name
      }
    }
  }
`
