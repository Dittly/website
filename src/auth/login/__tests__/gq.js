/* eslint-env jest */
import * as gqQueries from '../gq'

describe('src/auth/login', () => {
  Object.entries(gqQueries).forEach(([queryName, query]) => {
    it(`query "${queryName}" matches the snapshot`, () => {
      expect(query).toMatchSnapshot()
    })
  })
})
