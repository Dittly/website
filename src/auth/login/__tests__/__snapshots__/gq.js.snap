// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`src/auth/login query "signIn" matches the snapshot 1`] = `
Object {
  "definitions": Array [
    Object {
      "directives": Array [],
      "kind": "OperationDefinition",
      "name": Object {
        "kind": "Name",
        "value": "Signin",
      },
      "operation": "mutation",
      "selectionSet": Object {
        "kind": "SelectionSet",
        "selections": Array [
          Object {
            "alias": undefined,
            "arguments": Array [
              Object {
                "kind": "Argument",
                "name": Object {
                  "kind": "Name",
                  "value": "email",
                },
                "value": Object {
                  "fields": Array [
                    Object {
                      "kind": "ObjectField",
                      "name": Object {
                        "kind": "Name",
                        "value": "email",
                      },
                      "value": Object {
                        "kind": "Variable",
                        "name": Object {
                          "kind": "Name",
                          "value": "email",
                        },
                      },
                    },
                    Object {
                      "kind": "ObjectField",
                      "name": Object {
                        "kind": "Name",
                        "value": "password",
                      },
                      "value": Object {
                        "kind": "Variable",
                        "name": Object {
                          "kind": "Name",
                          "value": "password",
                        },
                      },
                    },
                  ],
                  "kind": "ObjectValue",
                },
              },
            ],
            "directives": Array [],
            "kind": "Field",
            "name": Object {
              "kind": "Name",
              "value": "signinUser",
            },
            "selectionSet": Object {
              "kind": "SelectionSet",
              "selections": Array [
                Object {
                  "alias": undefined,
                  "arguments": Array [],
                  "directives": Array [],
                  "kind": "Field",
                  "name": Object {
                    "kind": "Name",
                    "value": "token",
                  },
                  "selectionSet": undefined,
                },
              ],
            },
          },
        ],
      },
      "variableDefinitions": Array [
        Object {
          "defaultValue": undefined,
          "kind": "VariableDefinition",
          "type": Object {
            "kind": "NonNullType",
            "type": Object {
              "kind": "NamedType",
              "name": Object {
                "kind": "Name",
                "value": "String",
              },
            },
          },
          "variable": Object {
            "kind": "Variable",
            "name": Object {
              "kind": "Name",
              "value": "email",
            },
          },
        },
        Object {
          "defaultValue": undefined,
          "kind": "VariableDefinition",
          "type": Object {
            "kind": "NonNullType",
            "type": Object {
              "kind": "NamedType",
              "name": Object {
                "kind": "Name",
                "value": "String",
              },
            },
          },
          "variable": Object {
            "kind": "Variable",
            "name": Object {
              "kind": "Name",
              "value": "password",
            },
          },
        },
      ],
    },
  ],
  "kind": "Document",
  "loc": Object {
    "end": 144,
    "start": 0,
  },
}
`;
